### DISCUSSION TITLE
How does Gleam do static dispatch?

### DISCUSSION THREAD

**Discussion by KSXGitHub** [gleam-lang/gleam#2664] [Questions & support]
Something like a trait, a typeclass, or an interface.

I read the language tour and saw no mention of it.

**Comment by lpil** [comment]
Those are all (most often) dynamic dispatch, static dispatch is done through regular function calls.

**Reply by KSXGitHub** [reply]
I don't see any mention of dynamic dispatch in the tour either, or polymorphism in general.

**Reply by lpil** [reply]
Dynamic dispatch is somewhat unrelated to polymorphism. We support parametric polymorphism

**Reply by KSXGitHub** [reply]
Do the language team plan to add ad-hoc polymorphism?

**Reply by lpil** [reply]
No, here's some info: <https://gleam.run/frequently-asked-questions/#will-gleam-have-type-classes>
